pennylane/devices/default_qubit_jax.py

```diff
@@ -0,0 +1,193 @@
+    _dot = staticmethod(jnp.dot)
+    _abs = staticmethod(jnp.abs)
+    _reduce_sum = staticmethod(lambda array, axes: jnp.sum(array, axis=tuple(axes)))
+    _reshape = staticmethod(jnp.reshape)
+    _flatten = staticmethod(lambda array: array.ravel())
+    _gather = staticmethod(lambda array, indices: array[indices])
+    _einsum = staticmethod(jnp.einsum)
+    _cast = staticmethod(jnp.array)
+    _transpose = staticmethod(jnp.transpose)
+    _tensordot = staticmethod(
+        lambda a, b, axes: jnp.tensordot(
+            a, b, axes if isinstance(axes, int) else list(map(tuple, axes))
+        )
+    )
+    _conj = staticmethod(jnp.conj)
+    _imag = staticmethod(jnp.imag)
+    _roll = staticmethod(jnp.roll)
+    _stack = staticmethod(jnp.stack)
+
+    def __init__(self, wires, *, shots=1000, analytic=True, prng_key=None):
+        super().__init__(wires, shots=shots, analytic=analytic, cache=0)
+
+        # prevent using special apply methods for these gates due to slowdown in jax
+        # implementation
+        del self._apply_ops["PauliY"]
+        del self._apply_ops["Hadamard"]
+        del self._apply_ops["CZ"]
+        self._prng_key = prng_key
+
+    @classmethod
+    def capabilities(cls):
+        capabilities = super().capabilities().copy()
+        capabilities.update(
+            passthru_interface="jax",
+            supports_reversible_diff=False,
+        )
+        return capabilities
+
+    @staticmethod
+    def _scatter(indices, array, new_dimensions):
+        new_array = jnp.zeros(new_dimensions, dtype=array.dtype.type)
+        new_array = new_array.at[indices].set(array)
+        return new_array
+
+    def _get_unitary_matrix(self, unitary):
+        """Return the matrix representing a unitary operation.
+
+        Args:
+            unitary (~.Operation): a PennyLane unitary operation
+
+        Returns:
+            array[complex]: Returns a 2D matrix representation of
+            the unitary in the computational basis, or, in the case of a diagonal unitary,
+            a 1D array representing the matrix diagonal.
+        """
+        op_name = unitary.name
+        if op_name in self.parametric_ops:
+            if op_name == "MultiRZ":
+                return self.parametric_ops[unitary.name](*unitary.parameters, len(unitary.wires))
+            return self.parametric_ops[unitary.name](*unitary.parameters)
+
+        if isinstance(unitary, DiagonalOperation):
+            return unitary.eigvals
+
+        return unitary.matrix
+
+    def sample_basis_states(self, number_of_states, state_probability):
+        """Sample from the computational basis states based on the state
+        probability.
+
+        This is an auxiliary method to the generate_samples method.
+
+        Args:
+            number_of_states (int): the number of basis states to sample from
+
+        Returns:
+            List[int]: the sampled basis states
+        """
+        if self._prng_key is None:

```

Comment:
its a _little_ broken by sampling. I need to document it, but you have to do a little bit of fanciness to make it work correctly with `jax.jit`.

The main issue is if you try to `jit` qnode code, if you don't variably pass in a PRNGKey, it'll always output the same samples.

Examples:

This is "broken"
```python
@jax.jit
@qml.device("default.qubit.jax", interface="jax")
def circuit():
  # Some stuff....
  return qml.sample()

a = circuit()
b = circuit() # b will be the exact same samples as a.
```  

Whereas this would be the "correct" way of getting it to work

```python
@jax.jit
def keyed_circuit(key):
  @qml.device("default.qubit.jax", interface="jax", prng_key=key)
  def circuit():
    # Some stuff....
    return qml.sample()
  return circuit()

key1 = jax.random.PRNGKey(0)
key2 = jax.random.PRNGKey(1)
a = keyed_circuit(key1)
b = keyed_circuit(key2) # b will be different samples now.
```

Reply:
Might be worth adding this example to the docstring! We also have a `.. UsageDetails::` directive for adding more advanced usage examples that are collapsed by default.